src/app/cli/src/init/test_ledger_application.ml

```diff
@@ -218,7 +221,34 @@ let test ~privkey_path ~ledger_path ?prev_block_path ~first_partition_slots
   let stop_tracing =
     if tracing then (fun x -> Mina_tracing.stop () ; x) else ident
   in
+  let results = ref [] in
   let init_root = Ledger.merkle_root init_ledger in
+  let save_preparation_times time =
+    if Option.is_some benchmark then results := !results @ [ time ]
+  in
+  let save_and_dump_benchmarks final_time =
+    let calculate_mean results =
+      let preparation_steps = List.drop_last_exn results in
+      let sum = Float.of_int (List.length preparation_steps) in
+      printf !"sum: %f\n" sum ;
+      List.fold preparation_steps ~init:Float.zero ~f:(fun acc time ->
```

Comment:
I suggest that we just add times up in the `List.fold` and divide by number of preparation steps at the very end. This is slightly more intuitive for taking an average